src/common/vncauth.c

```diff
@@ -102,7 +102,10 @@ rfbEncryptAndStorePasswd(char *passwd, char *fname)
 
     /* Do encryption in-place - this way we overwrite our copy of the plaintext
        password */
-    encrypt_rfbdes(encryptedPasswd, &out_len, fixedkey, encryptedPasswd, sizeof(encryptedPasswd));
+    if (encrypt_rfbdes(encryptedPasswd, &out_len, fixedkey, encryptedPasswd, sizeof(encryptedPasswd)) == 0) {
```

Comment:
There are a few more callers, shouldn't this get checked there as well?

Reply:
Checked in all callers now.

Reply:
Thanks! Fundamental question: what did happen without this check in an error case - wouldn't it fail in a later stage anyway?

Reply:
The `storepasswd` tool would silently write the unencrypted password into the given file.